src/cljs/commiteth/bounties.cljs

```diff
@@ -4,51 +4,95 @@
+                                      issue-url
+                                      pull-request-url]]
             [commiteth.handlers :as handlers]
             [commiteth.db :as db]
             [commiteth.ui-model :as ui-model]
             [commiteth.subscriptions :as subs]
             [commiteth.util :as util]))
 
+(defn relevant-activity-item [items]
+  [:div
+   (for [item items]
+     (let [{:keys [avatar-url
+                   pr-title
+                   updated
+                   repo-owner
+                   repo-name
+                   pr-number]} item]
+       ^{:key (random-uuid)}
+       [:div.bounty-claims-row.open-bounty-item-content
+        [:div.bounty-claims-icon
+         [:div.ui.tiny.circular.image
+          [:img {:src avatar-url}]]]
+        [:span.bounty-claims-text
+         [:a
+          {:href (pull-request-url repo-owner repo-name pr-number)}
+          pr-title]
+         [:span.time.bounty-claims-time (human-time updated)]]]))])
+
+(defn find-relevant-activity [issue-number]
+  (let [activity-feed     (rf/subscribe [:activity-feed])
+        relevant-activity (filter #(= issue-number (:issue-number %))
+                                  @activity-feed)]
+    (when relevant-activity
+      (relevant-activity-item relevant-activity))))
 
 (defn bounty-item [bounty]
   (let [{avatar-url   :repo_owner_avatar_url
          owner        :repo-owner
          repo-name    :repo-name
          issue-title  :issue-title
          issue-number :issue-number
+         issue-id     :issue-id
          updated      :updated
          tokens       :tokens
          balance-eth  :balance-eth
          value-usd    :value-usd
          claim-count  :claim-count} bounty
-        full-repo  (str owner "/" repo-name)
-        repo-url   (str "https://github.com/" full-repo)
-        repo-link  [:a {:href repo-url} full-repo]
-        issue-link [:a
-                    {:href (issue-url owner repo-name issue-number)}
-                    issue-title]]
-    [:div.open-bounty-item
-     [:div.open-bounty-item-content
-      [:div.header issue-link]
-      [:div.bounty-item-row
-       [:div.time (human-time updated)]
-       [:span.bounty-repo-label repo-link]]
+        full-repo                   (str owner "/" repo-name)
+        repo-url                    (str "https://github.com/" full-repo)
+        repo-link                   [:a {:href repo-url} full-repo]
+        issue-link                  [:a
+                                     {:href (issue-url owner repo-name issue-number)}
+                                     issue-title]
+        open-claims-click           (fn []
+                                      (rf/dispatch [::handlers/open-bounty-claim issue-id]))
+        close-claims-click          (fn []
+                                      (rf/dispatch [::handlers/close-bounty-claim issue-id]))
+        matches-current-issue?      (some #{issue-id} @(rf/subscribe [::subs/open-bounty-claims]))]
```

Comment:
Few things here:
1. When using subscriptions you should always wrap the actual rendering in an extra function. If you don't the subscription will be recreated on every render. See [Form-2 Components](https://github.com/reagent-project/reagent/blob/master/docs/CreatingReagentComponents.md#form-2--a-function-returning-a-function) in the Reagent/re-frame docs. The example has an atom but this equally applies to subscriptions. This also applies to how you're subscribing to `:activity-feed` in the `find-relevant-activity` function.
2. Creating a subscription for every item rendered isn't ideal and maybe one could argue that it adds too much logic to views (although clearly we're not aiming for logic-free views). A better approach might be to modify the bounties subscription to take into account the list of open bounty claims and add that information to each bounty. Similarly the list of claims could be injected into the bounty map this way. Once that is done you can just pass the data to a pure-ish component (rendered DOM entirely defined by function args).

Let me know if this makes sense.

Reply:
Thanks Martin! `2.` makes total sense and agreed could benefit from a refactor. I'll go through the documentation you linked to for reagent regarding `1.` as well 👍

Reply:
@martinklepsch This all worked really well to clean up the view logic (and in some cases just remove unnecessary code), so thank you for that. 

I just wanted to flag the decision over where to "pack in" the claims data in the subscriptions. There are both `:open-bounties` and `:open-bounties-page` subscriptions, and I opted to roll the claims into the `:open-bounties-page` subscription because it seemed concerned with generating a very UI-centric cut of the data, whereas `:open-bounties` struck me as generic enough that I didn't want to cram it with claims data if that subscription would be shuttled around for more general purposes. Does that logic seem sound? Happy to refactor if we want it to go elsewhere.